lib/puppet/defaults.rb

```diff
@@ -1209,11 +1211,17 @@ def self.valid_file_checksum_types
   define_settings(:master,
     :user => {
       :default    => "puppet",
-      :desc       => "The user puppet master should run as.",
+      :desc       => "The user puppet master should run as. Unused,
+      Puppet Server utilizes a system's service framework to run as
+      this user.",
+      :deprecated  => :completely,
```

Comment:
I'm don't think we can deprecate `user` and `group` quite yet, because the host's cert/key (not the CA stuff) is still shared between the `puppet-agent` and `puppet-server` packages. So puppet needs to know which `user` and `group` to map the `service` account to, so that puppetserver can read those files when running as `puppet` or `pe-puppet`.

Reply:
Grumble... Good catch, with our recent work with `puppetserver ca create` I wasn't thinking about the use case of using an agent to bootstrap a compile master's PKI.

You mind if I update the description and hook to disallow values besides `puppet` and `pe-puppet` and call out that, if set, this must be the value that matches their kind of installation (commercial or open source)?

Reply:
I think I'd prefer to leave out the deprecation. The `user` and `group` settings are typically used to share ssl files, but it's also used to share `server_datadir`, `report_dir`, ... While puppetserver has a separate `vardir` by default, it's configurable, so I'm not sure what workflows would be affected.